### PR TITLE
saw: init at 0.2.2

### PR DESCRIPTION
**Pull request by terlar**
###### Description of changes

Add package [saw](https://github.com/TylerBrock/saw) - a fast, multi-purpose tool for AWS CloudWatch Logs.

I guess it is a bit similar to the already packaged [awslogs](https://github.com/jorgebastida/awslogs). I have not tried that one yet, so I am not fully aware of the differences.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by papojari** [comment]
> I guess it is a bit similar to the already packaged [awslogs](https://github.com/jorgebastida/awslogs). I have not tried that one yet, so I am not fully aware of the differences.

Well, you should.

File: pkgs/tools/admin/saw/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  vendorSha256 = "2GTa5BEls5T1k7qQSpi6VprcTrjBmTKz7Ra5OsAJBFA=";
+
+  passthru.tests.version = testVersion {
+    package = saw;
+    command = "saw version";
+    version = "v${version}";
+  };
+
+  meta = with lib; {
+    description = "Fast, multi-purpose tool for AWS CloudWatch Logs.";
```
Comment (papojari) [inline pkgs/tools/admin/saw/default.nix:31]:
```suggestion
    description = "Fast, multi-purpose tool for AWS CloudWatch Logs";
```

**Review by auroraanna — CHANGES_REQUESTED** [review]

**Comment by papojari** [comment]
Result of `nixpkgs-review pr 166956` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>saw</li>
  </ul>
</details>

**Comment by terlar** [comment]
So apparently the creator of saw was a dissatisfied user of awslogs, perhaps in the words of the author as well as some commentors:
https://stackoverflow.com/questions/34018931/how-to-view-aws-log-real-time-like-tail-f#answer-50916992

Seems the UX and execution of saw is a bit nicer, and that has been my experience as well as I briefly explored awslogs now.

I guess these tools become less relevant with the updated awscli v2. But they still seem to have some advantages. Will explore this as well.

Addressed your comments.

**Review by auroraanna — APPROVED** [review]

**Comment by papojari** [comment]
It is done, right?

**Comment by papojari** [comment]
It is done, right?

**Comment by terlar** [comment]
Yes, it is slightly different. A bit nicer UX and potentially faster, but the true bottleneck is CloudWatch itself.

Yes, it is done. Good to go.

**Comment by jiegec** [comment]
`testVersion` should be replaced by `testers.testVersion`.

**Pull request closed without merging**